libcnb-test/src/container_port_mapping.rs

```diff
@@ -0,0 +1,228 @@
+        .clone()
+        .unwrap_or_default()
+        .first()
+        .cloned()
+        .ok_or(PortMapParseError::NoBindingForPort(container_port))?;
+
+    let host_address = port_binding
+        .host_ip
+        .ok_or(PortMapParseError::MissingHostIpAddress)
+        .and_then(|host_ip| {
+            host_ip
+                .parse()
+                .map_err(PortMapParseError::HostIpAddressParseError)
+        })?;
+
+    let host_port = port_binding
+        .host_port
+        .ok_or(PortMapParseError::MissingHostPort)
+        .and_then(|host_port| {
+            host_port
+                .parse()
+                .map_err(PortMapParseError::HostPortParseError)
+        })?;
+
+    Ok((container_port, SocketAddr::new(host_address, host_port)))
+}
+
+/// Create a new Bollard container config with the given exposed ports.
+///
+/// The exposed ports will be forwarded to random ports on the host.
+pub(crate) fn port_mapped_container_config(ports: &[u16]) -> bollard::container::Config<String> {
+    Config {
+        host_config: Some(HostConfig {
+            port_bindings: Some(
+                ports
+                    .iter()
+                    .map(|port| {
+                        (
+                            format!("{}/tcp", port),
+                            Some(vec![PortBinding {
+                                host_ip: None,
+                                host_port: None,
+                            }]),
+                        )
+                    })
+                    .collect::<PortMap>(),
+            ),
+            ..Default::default()
+        }),
+        exposed_ports: Some(
+            ports
+                .iter()
+                .map(|port| {
+                    (
+                        format!("{}/tcp", port),
+                        #[allow(clippy::zero_sized_map_values)]
+                        HashMap::new(), // Bollard requires this zero sized value map,
+                    )
+                })
+                .collect(),
+        ),
+        ..Default::default()
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use bollard::models::PortMap;
+
+    #[test]
+    fn parse_port_map_simple() {
+        let mut port_map: PortMap = HashMap::new();
+        port_map.insert(
+            String::from("12345/tcp"),
+            Some(vec![PortBinding {
+                host_ip: Some(String::from("0.0.0.0")),
+                host_port: Some(String::from("57233")),
+            }]),
+        );
+
+        port_map.insert(
+            String::from("80/tcp"),
+            Some(vec![PortBinding {
+                host_ip: Some(String::from("127.0.0.1")),
+                host_port: Some(String::from("51039")),
+            }]),
+        );
+
+        let mut expected_result: HashMap<u16, SocketAddr> = HashMap::new();
+        expected_result.insert(12345, "0.0.0.0:57233".parse().unwrap());
+        expected_result.insert(80, "127.0.0.1:51039".parse().unwrap());
```

Comment:
I don't know which is more idiomatic, but another way of writing this is:

```rust
        let expected_result: HashMap<u16, SocketAddr> = HashMap::from([
            (12345, "0.0.0.0:57233".parse().unwrap()),
            (80, "127.0.0.1:51039".parse().unwrap()),
        ]);
```

Reply:
Less mutability is always better in my book. :)